lib/src/component/abstract_transition.dart

```diff
@@ -184,7 +187,14 @@ abstract class AbstractTransitionComponent<T extends AbstractTransitionProps, S
       skipCount = 0;
     }
 
+    var timer = new Timer(transitionTimeout, () {
+      assert(ValidationUtil.warn(
+        'The number of transitions expected to complete have not completed. Something is most likely wrong.'
+      ));
```

Comment:
Mm, yes, we should, in order to account for when the warning isn't caught by the time this is deployed